tiatoolbox/tools/tissuemask.py

```diff
@@ -0,0 +1,225 @@
+                e.g. regions of tissue vs background.
+        """
+        if not self.fitted:
+            raise Exception("Fit must be called before transform.")
+
+    def fit_transform(self, image: np.ndarray, **fit_params) -> np.ndarray:
+        """Apply :func:`fit` and :func:`transform` in one step.
+
+        Sometimes it can be more optimal to perform both at the same
+        time for a single sample. In this case the base implementation
+        of :fun:`fit` followed by :func:`transform` can be overridden.
+
+        Args:
+            image (:class:`numpy.ndarray`): Image to create mask from.
+            fit_params (dict): Generic key word arguments passed to fit.
+        """
+        self.fit(image, **fit_params)
+        return self.transform(image)
+
+
+class OtsuTissueMasker(TissueMasker):
+    """Tissue masker which uses Otsu's method to determine background."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.threshold = None
+
+    def fit(self, image: np.ndarray, mask=None) -> None:
+        # find Otsu's threshold
+        if len(image.shape) == 3:
+            gray = cv2.cvtColor(image, cv2.COLOR_RGB2GRAY)
+        else:
+            gray = image
+
+        self.threshold, _ = cv2.threshold(
+            gray, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU
+        )
+
+        self.fitted = True
+
+    def transform(self, image: np.ndarray) -> np.ndarray:
+        super().transform(image)
+
+        if len(image.shape) == 3:
+            gray = cv2.cvtColor(image, cv2.COLOR_RGB2GRAY)
+        else:
+            gray = image
+
+        mask = gray < self.threshold
+
+        return mask.astype(bool)
+
+
+class MorphologicalMasker(TissueMasker):
+    """Tissue masker which uses a threshold and simple morphological operations.
+
+    This method applies Otsu's threshold before a simple small region
+    removal, followed by a morphological dilation. The kernel for the
+    dilation is an ellipse of radius 64/mpp unless a value is given for
+    kernel_size to :func:`fit`. MPP is estimated from objective power
+    via func:`tiatoolbox.utils.misc.objective_power2mpp` if a power
+    argument is given instead of mpp to the initialiser.
+    For small region removal, the minimum area size defaults to the area
+    of the kernel.
+    If the scale of the morphological operations can also be manually
+    specified with the `kernel_size` argument, for example if the
+    automatic scale from mpp or objective power is too large or small.
+    """
+
+    def __init__(
+        self, *, mpp=None, power=None, kernel_size=None, min_region_size=None
+    ) -> None:
+        """Initialise a morphological masker.
+
+        Args:
+            mpp (float or tuple of float):
+                The microns per-pixel of the image to be masked. Used to
+                calculate kernel_size a 64/mpp, optional.
+            power (float or tuple of float):
```

Comment:
please use `tuple(float)` instead of tuple of float for consistency with the rest of repo